source/extensions/filters/http/header_to_metadata/header_to_metadata_filter.cc

```diff
@@ -100,16 +101,18 @@ bool HeaderToMetadataFilter::addMetadata(StructMap& map, const std::string& meta
   // Sane enough, add the key/value.
   switch (type) {
   case envoy::config::filter::http::header_to_metadata::v2::Config_ValueType_STRING:
-    val.set_string_value(value);
+    val.set_string_value(std::string(value));
```

Comment:
nit: shouldn't we use `ProtobufTypes::String` ([definition](https://github.com/envoyproxy/envoy/blob/c34f974439536072d0377bf8f10047489bc3a94b/source/common/protobuf/protobuf.h#L49)) here?

Reply:
yes, done.  good catch.